datafusion/core/tests/custom_sources.rs

```diff
@@ -296,3 +300,77 @@ fn contains_empty_exec(plan: Arc<dyn ExecutionPlan>) -> bool {
         contains_empty_exec(Arc::clone(&plan.children()[0]))
     }
 }
+
+#[tokio::test]
+async fn test_get_scan_files() -> Result<()> {
+    let session_ctx = SessionContext::new();
+    let testdata = parquet_test_data();
+    let path = format!("{testdata}/alltypes_plain.parquet");
+    let options = ParquetReadOptions::new();
+    session_ctx
+        .register_parquet("t", &path, options)
+        .await
+        .unwrap();
+
+    let exec = session_ctx
+        .sql("select * from t")
+        .await
+        .unwrap()
+        .create_physical_plan()
+        .await
+        .unwrap();
+
+    // test PartitionedFileFinder API
+    let scan_files = PartitionedFileFinder::new().find(exec.clone());
+    validate_name(scan_files, "alltypes_plain.parquet");
+
+    // test get_scan_files API
+    validate_name(get_scan_files(exec).unwrap(), "alltypes_plain.parquet");
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_get_scan_files_custom() -> Result<()> {
+    let ctx = SessionContext::new();
+
+    let exec = ctx
+        .read_table(Arc::new(CustomTableProvider))
+        .unwrap()
+        .create_physical_plan()
+        .await
+        .unwrap();
+
+    let scan_files = PartitionedFileFinder::new().find(exec.clone());
+    assert!(scan_files.is_empty());
+
+    // with custom finder
+    let scan_files = PartitionedFileFinder::new()
+        .with_finder(|exec| {
```

Comment:
here is an example of how to find files using a custom datasource / plan -- cc @not-my-profile